src/common/redis_module/ray_redis_module.c

```diff
@@ -272,6 +273,186 @@ int ResultTableLookup_RedisCommand(RedisModuleCtx *ctx,
+ * with the same task ID.
+ *
+ * This is called from a client with the command:
+ *
+ *     RAY.task_table_add <task ID> <state> <node ID> <task spec>
+ *
+ * @param task_id A string that is the ID of the task.
+ * @param state A string that is the current scheduling state (a
+ *        scheduling_state enum instance).
+ * @param node_id A string that is the ID of the associated node, if any.
+ * @param task_spec A string that is the specification of the task, which can
+ *        be cast to a `task_spec`.
+ * @return OK if the operation was successful.
+ */
+int TaskTableAddTask_RedisCommand(RedisModuleCtx *ctx,
+                                  RedisModuleString **argv,
+                                  int argc) {
+  if (argc != 5) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  return TaskTableWrite_RedisCommand(ctx, argv[1], argv[2], argv[3], argv[4]);
+}
+
+/**
+ * Update an entry in the task table. This does not update the task
+ * specification in the table.
+ *
+ * This is called from a client with the command:
+ *
+ *     RAY.task_table_update_task <task ID> <state> <node ID>
+ *
+ * @param task_string A string which is the task instance.
+ * @return OK if the operation was successful.
+ */
+int TaskTableUpdate_RedisCommand(RedisModuleCtx *ctx,
+                                 RedisModuleString **argv,
+                                 int argc) {
+  if (argc != 4) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  return TaskTableWrite_RedisCommand(ctx, argv[1], argv[2], argv[3], NULL);
+}
+
+/**
+ * Get an entry from the task table.
+ *
+ * This is called from a client with the command:
+ *
+ *     RAY.task_table_get <task ID>
+ *
+ * @param task_id A string of the task ID to look up.
+ * @return An array of strings representing the task fields in the following
+ *         order: 1) (integer) scheduling state 2) (string) associated node ID,
+ *         if any 3) (string) the task specification, which can be casted to a
+ *         task_spec. If the task ID is not in the table, returns nil.
+ */
+int TaskTableGetTask_RedisCommand(RedisModuleCtx *ctx,
+                                  RedisModuleString **argv,
+                                  int argc) {
+  if (argc != 2) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  RedisModuleKey *key =
+      OpenPrefixedKey(ctx, TASK_PREFIX, argv[1], REDISMODULE_READ);
+
+  int keytype = RedisModule_KeyType(key);
+  if (keytype != REDISMODULE_KEYTYPE_EMPTY) {
+    /* If the key exists, look up the fields and return them in an array. */
+    RedisModuleString *state, *node, *task_spec;
+    RedisModule_HashGet(key, REDISMODULE_HASH_CFIELDS, "state", &state, "node",
+                        &node, "spec", &task_spec, NULL);
+    if (state == NULL || node == NULL || task_spec == NULL) {
+      /* We must have either all fields or no fields. */
+      return RedisModule_ReplyWithError(
+          ctx, "Missing fields in the task table entry");
+    }
+
+    size_t state_length;
```

Comment:
this is a nit pick, but uninitialized stack variables filled in by external libraries are a bad idea. I've seen many cases where calls to external libraries may fail before they write a value into this pointer. Non-determinism follows.